ui/lib/kv/addon/components/page/secret/version-history.hbs

```diff
@@ -1,11 +1,102 @@
+      @queryParams={{hash version=versionData.version}}
+      data-test-version-linked-block={{versionData.version}}
+    >
+      <div class="level is-mobile">
+        <div class="is-grid is-grid-3-columns is-three-fourths-width">
+          {{! version number and icon }}
+          <div class="align-self-center">
+            <Icon @name="history" class="has-text-grey" data-test-version />
+            <span class="has-text-weight-semibold has-text-black">
+              Version
+              {{versionData.version}}
+            </span>
+          </div>
+          {{! icons }}
+          <div class="align-self-center" data-test-icon-holder={{versionData.version}}>
+            {{#if (loose-equal versionData.version @metadata.currentVersion)}}
+              <div>
+                <span class="has-text-success is-size-8 is-block">
+                  <Icon @name="check-circle-fill" />Current
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.deletion_time}}
+              <div>
+                <span class="has-text-grey is-size-8 is-block">
+                  <Icon @name="x-square-fill" />Deleted
+                </span>
+              </div>
+            {{/if}}
+            {{#if versionData.destroyed}}
+              <div>
+                <span class="has-text-danger is-size-8 is-block">
+                  <Icon @name="x-square-fill" />Destroyed
+                </span>
+              </div>
+            {{/if}}
+          </div>
+          {{! version created date }}
+          <div class="is-size-8 has-text-weight-semibold has-text-grey align-self-center">
+            <ToolTip @verticalPosition="above" @horizontalPosition="center" as |T|>
+              <T.Trigger data-test-tooltip-trigger tabindex="-1">
+                Created
+                {{date-format versionData.created_time "MMM dd, yyyy hh:mm a"}}
+              </T.Trigger>
+              <T.Content @defaultClass="tool-tip smaller-font">
+                <div class="box" data-test-hover-copy-tooltip-text>
+                  {{versionData.created_time}}
+                </div>
+              </T.Content>
+            </ToolTip>
+          </div>
+        </div>
 
-<hr class="is-marginless has-background-gray-200" />
\ No newline at end of file
+        <div class="level-right">
+          <div class="level-item">
+            <PopupMenu @name="version-{{versionData.version}}">
+              <nav class="menu">
+                <ul class="menu-list">
+                  <li>
+                    <LinkTo @route="secret.details" @model={{@metadata.path}} @query={{hash version=versionData.version}}>
+                      View version
+                      {{versionData.version}}
+                    </LinkTo>
+                  </li>
+                  <li>
+                    <LinkTo
+                      @route="secret.details.edit"
+                      @model={{@metadata.path}}
```

Comment:
Same question here about pathing`@path` instead. And should this be wrapped in a conditional for whether or not a user has update data permission?

Reply:
Re: permissions. Good catch. We didn't have a conditional for permissions in the old view, but I can't think through a reason why we shouldn't now. I'll amend.